tests/python_package_test/test_engine.py

```diff
@@ -1508,6 +1508,50 @@ def test_parameters_are_loaded_from_model_file(tmp_path, capsys, rng):
     np.testing.assert_allclose(preds, orig_preds)
 
 
+def test_string_serialized_params_retrieval(rng):
+    # Random train data
+    train_x = rng.random((500, 3))
+    train_y = rng.integers(0, 1, 500)
+    train_data = lgb.Dataset(train_x, train_y)
+
+    # Parameters
+    params = {
+        "boosting": "gbdt",
+        "deterministic": True,
+        "feature_contri": [0.5] * train_x.shape[1],
+        "interaction_constraints": [[0, 1], [0]],
+        "objective": "binary",
+        "metric": ["auc"],
+        "num_leaves": 7,
+        "learning_rate": 0.05,
+        "feature_fraction": 0.9,
+        "bagging_fraction": 0.8,
+        "bagging_freq": 5,
+        "verbosity": -100,
+    }
+
+    # train a model and serialize it to a string in memory
+    model = lgb.train(params, train_data, num_boost_round=2)
+    model_serialized = model.model_to_string()
+
+    # load a new model with the string
+    with pytest.warns(UserWarning, match="Ignoring params argument, using parameters from model string."):
+        new_model = lgb.Booster(params={"num_leaves": 7}, model_str=model_serialized)
```

Comment:
I think it'd be helpful to set a different value here to make sure that it's actually ignored, e.g. `"num_leaves": 35`

Reply:
Done